ratatui-core/src/backend/test.rs

```diff
@@ -67,6 +68,225 @@ fn buffer_view(buffer: &Buffer) -> String {
+                        let style_start = format_style_start(&c.style());
+                        if !style_start.is_empty() {
+                            view.push_str(&style_start);
+                        }
+                    }
+                    current_style = c.style();
+                }
+                view.push_str(c.symbol());
+            } else {
+                overwritten.push((x, c.symbol()));
+            }
+            skip = core::cmp::max(skip, c.symbol().width()).saturating_sub(1);
+        }
+        // Close any remaining style tags at the end of the line
+        if current_style != crate::style::Style::default() {
+            let style_end = format_style_end(&current_style);
+            if !style_end.is_empty() {
+                view.push_str(&style_end);
+            }
+        }
+        view.push('"');
+        if !overwritten.is_empty() {
+            write!(&mut view, " Hidden by multi-width symbols: {overwritten:?}").unwrap();
+        }
+        view.push('\n');
+    }
+    view
+}
+
+/// Format the start of a style tag based on the cell's style
+fn format_style_start(style: &crate::style::Style) -> String {
+    let mut tags = Vec::new();
+
+    // Add foreground color
+    if let Some(fg) = &style.fg {
+        if *fg != crate::style::Color::Reset {
+            tags.push(format_color(*fg));
+        }
+    }
+
+    // Add background color
+    if let Some(bg) = &style.bg {
+        if *bg != crate::style::Color::Reset {
+            tags.push(format!("bg:{}", format_color(*bg)));
+        }
+    }
+
+    // Add modifiers
+    if style.add_modifier.contains(crate::style::Modifier::BOLD) {
+        tags.push("bold".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::ITALIC) {
+        tags.push("italic".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::UNDERLINED)
+    {
+        tags.push("underlined".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::DIM) {
+        tags.push("dim".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::CROSSED_OUT)
+    {
+        tags.push("crossed_out".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::SLOW_BLINK)
+    {
+        tags.push("slow_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::RAPID_BLINK)
+    {
+        tags.push("rapid_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::REVERSED)
+    {
+        tags.push("reversed".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::HIDDEN) {
+        tags.push("hidden".to_string());
+    }
+
+    if tags.is_empty() {
+        String::new()
+    } else {
+        format!("{{{}}}", tags.join(","))
+    }
+}
```

Comment:
I think this needs syntax still needs some discussion before implementing it. I see that we have mostly agreed on this style in #1402 but #493 has some really good arguments for a more general-purpose use for it. In that case it would mean that we need to have a parser for this syntax first (doing the opposite of what's being done here) and initially support this in `ratatui-macros`' e.g. `line!` macro.

So I would refrain from adding this syntax here and first add a proper parser for it.

Please see the discussion in #493 and example implementations such as [`tui-markup`](https://github.com/7sDream/tui-markup).